docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,272 @@
+used for the microVM state file as it contains internal state structures for device
+emulation, vCPUs and others that can change their format from a Firecracker version
+to another. Block devices and memory files do not support versioning. You have to
+handle the separate block device file components of the snapshot.
+
+- _on failure_: no side-effects.
+
+### Creating diff snapshots
+
+For creating a diff snapshot, the same API command should be used, but with
+`snapshot_type` field set to `Diff`.
+*Note*: If omitted from the request body, `snapshot_type` default value is `Full`.
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/create" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_type\": \"Diff\",
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+**Prerequisites**: It can be called only when the microVM is `Paused`.
+**Effects**:
+- _on success_:
+The file indicated by `/path/to/snapshot_file` contains the devices' model state and
+emulation state, same as when creating a full snapshot. The one indicated by
+`/path/to/mem_file` contains this time a **diff copy** of the guest memory - the diff
+is consisted of the memory pages which have been dirtied since the last snapshot creation
+or since the creation of the microVM, whichever of these events was the most recent.
+All the other effects mentioned in the **Effects** section from `### Creating full snapshots`
+apply here.
+- _on failure_: no side-effects.
+
+*Note*: To be able to do any incremental (diff) snapshots, you must set `track_dirty_pages`
+field to `true` when configuring the `/machine-config` resource. Example:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i  \
+    -X PUT 'http://localhost/machine-config' \
+    -H 'Accept: application/json'            \
+    -H 'Content-Type: application/json'      \
+    -d '{
+            "vcpu_count": 2,
+            "mem_size_mib": 1024,
+            "ht_enabled": false,
+            "track_dirty_pages": true
+    }'
+```
+
+Enabling this support enables KVM dirty page tracking, so it comes at a cost; you should
+only use it when needed.
+
+Creating a snapshot will **not** influence state, will **not** stop or end the microVM,
+it can be used as before, so you can resume the microVM if you still want to use it.
+
+Resuming the microVM can be done by sending the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/vm" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Resumed"
+    }"
+```
+
+**Prerequisites**: You can resume the microVM any time after the microVM is booted.
+                   Successive calls of this request keep the microVM in the `Resumed`
```

Comment:
This is not clear. The resume call does something only if the microvm is paused otherwise it is ignored.

Reply:
I rephrased it.